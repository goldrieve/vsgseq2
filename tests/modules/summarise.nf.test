nextflow_process {

    name "Test Process SUMMARISE"
    script "modules/summarise.nf"
    process "SUMMARISE"

    test("Should run summary steps") {

        when {
            params {
                threshold = 100000
            }
            process {
                """
                input[0] = "${projectDir}/tests/data/early_1_quant/, ${projectDir}/tests/data/late_1_quant/"
                input[1] = params.threshold
                input[2] = "${projectDir}/tests/data/latesingle_ORF_VSGs.fasta"
                input[3] = "${projectDir}/tests/data/VSGome.fasta.clstr"
                input[4] = "${projectDir}/tests/data/VSGome.fasta"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
