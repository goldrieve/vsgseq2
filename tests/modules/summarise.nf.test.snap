{
    "Should run summary steps": {
        "content": [
            {
                "0": [
                    "tpm.csv:md5,3d1e151e6f0e65ad6e1d0b6e280d19dd"
                ],
                "1": [
                    "num_reads.csv:md5,d6a043ce56aa3690e74130cd1a041b5a"
                ],
                "10": [
                    "champion_vsgs.fasta:md5,e61f4e60eede6401fdfb415390b08f4c"
                ],
                "2": [
                    "total_read_counts.csv:md5,28bfe7acad8c713c61bc6005347dc757"
                ],
                "3": [
                    "vsg_count.csv:md5,c2059fe15d1fca1498e299ce7c6c6d9d"
                ],
                "4": [
                    "filtered_tpm.csv:md5,3d1e151e6f0e65ad6e1d0b6e280d19dd"
                ],
                "5": [
                    "filtered_tpm_clusters.csv:md5,0e5141df0a92c43f9e3f2d86e5b85411"
                ],
                "6": [
                    "length.csv:md5,a512692df3488283c7ff83214cf106c5"
                ],
                "7": [
                    "filtered_tpm_clusters_length.csv:md5,5142a3d772e66f98bb51e4663222c152"
                ],
                "8": [
                    "cluster_tpm.csv:md5,23b74fa576e97dfc1456d3dc637c6e74"
                ],
                "9": [
                    "cluster_champion.csv:md5,082444ebbf799633611b4c55b561ecde"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-29T12:44:45.555194108"
    }
}